workflows/pipe-common/shell/sge_epilog

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# Copyright (c) 2015 Kota Yamaguchi
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+# 3. The name of the author may not be used to endorse or promote products
+#    derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+# OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+# NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+# THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+echo "Executing Cloud Pipeline epilog script as $(whoami)..."
+
+JOB_ENV_FILE="$SGE_JOB_SPOOL_DIR/environment"
+
+if [ ! -f "$JOB_ENV_FILE" ]; then
+    echo "ERROR: Job environment file is not accessible at $JOB_ENV_FILE. Exiting..."
+    exit 100
+fi
+
+CP_CAP_GE_CONSUMABLE_RESOURCE_NAME_GPU="${CP_CAP_GE_CONSUMABLE_RESOURCE_NAME_GPU:-gpus}"
+CP_CAP_GE_CONSUMABLE_RESOURCE_LOCKS_GPU="${CP_CAP_GE_CONSUMABLE_RESOURCE_LOCKS_GPU:-/tmp/sge-gpus-locks}"
+
+JOB_GPUS_NUMBER="$(qstat -j "$JOB_ID" | grep -oP "$CP_CAP_GE_CONSUMABLE_RESOURCE_NAME_GPU=\K\d+")"
+JOB_GPUS_NUMBER="${JOB_GPUS_NUMBER:-0}"
+
+if [ "$JOB_GPUS_NUMBER" -le 0 ]; then
+    exit 0
+fi
+
+JOB_GPU_DEVICES=$(grep -oP "CUDA_VISIBLE_DEVICES=\K.*" "$JOB_ENV_FILE" | sed -e "s/,/ /g")
+for GPU_DEVICE in $JOB_GPU_DEVICES; do
+    echo -n "Releasing GPU device $GPU_DEVICE... "
+    if ! rmdir "$CP_CAP_GE_CONSUMABLE_RESOURCE_LOCKS_GPU/inuse/$GPU_DEVICE" 2>/dev/null; then
```

Comment:
in which cases we will not be able to delete it? Can we forcefully delete directory here? 
I worry about the case when for some reason we will not remove it and next jobs will fail because of `synthetic` insufficient GPUs.

Reply:
Generally, it is a very case related to prolog and epilog scripts misconfiguration and I think it is quite ok to have such a warning. As I see it, most likely, it can be a permission level issue. In that case forceful deletion won't help but we will still have the warning for troubleshooting. So I would propose to keep it as it is.

Reply:
Got it. I agree